src/oidcendpoint/oidc/authorization.py

```diff
@@ -123,601 +98,41 @@ class Authorization(Endpoint):
-            if re_authenticate(request, authn):
-                # demand re-authentication
-                return {"function": authn, "args": authn_args}
-            else:
-                # I get back a dictionary
-                user = identity["uid"]
-                if "req_user" in kwargs:
-                    sids = self.endpoint_context.sdb.get_sids_by_sub(kwargs["req_user"])
-                    if (
-                            sids
-                            and user
-                            != self.endpoint_context.sdb.get_authentication_event(
-                        sids[-1]
-                    ).uid
-                    ):
-                        logger.debug("Wanted to be someone else!")
-                        if "prompt" in request and "none" in request["prompt"]:
-                            # Need to authenticate but not allowed
-                            return {
-                                "error": "login_required",
-                                "return_uri": redirect_uri,
-                            }
-                        else:
-                            return {"function": authn, "args": authn_args}
-
-        authn_event = None
-        if session:
-            authn_event = session.get('authn_event')
-
-        if authn_event is None:
-            authn_event = create_authn_event(
-                identity["uid"],
-                identity.get("salt", ""),
-                authn_info=authn_class_ref,
-                time_stamp=_ts,
-            )
-
-        _exp_in = authn.kwargs.get("expires_in")
-        if _exp_in and "valid_until" in authn_event:
-            authn_event["valid_until"] = utc_time_sans_frac() + _exp_in
-
-        return {"authn_event": authn_event, "identity": identity, "user": user}
-
-    def create_authn_response(self, request, sid):
-        """
-
-        :param self:
-        :param request:
-        :param sid:
-        :return:
-        """
-        # create the response
-        aresp = self.response_cls()
-        if request.get("state"):
-            aresp["state"] = request["state"]
-
-        if "response_type" in request and request["response_type"] == ["none"]:
-            fragment_enc = False
-        else:
-            _context = self.endpoint_context
-            _sinfo = _context.sdb[sid]
-
-            if request.get("scope"):
-                aresp["scope"] = request["scope"]
-
-            rtype = set(request["response_type"][:])
-            handled_response_type = []
-
-            fragment_enc = True
-            if len(rtype) == 1 and "code" in rtype:
-                fragment_enc = False
 
-            if "code" in request["response_type"]:
-                _code = aresp["code"] = _context.sdb[sid]["code"]
-                handled_response_type.append("code")
-            else:
-                _context.sdb.update(sid, code=None)
-                _code = None
+    def setup_client_session(self, user_id: str, request: dict) -> str:
```

Comment:
Perhaps we should allow the caller to pass his own `sector_identifier_uri` and `salt` to allow for greater flexibility

Reply:
`sector_identifier_uri` comes from the authorization request and should not be entered any other way.